src/gui/graphview.cpp

```diff
@@ -322,31 +322,46 @@ void GraphView::drawSelectedStars(QPainter& painter, double nodeRadius) const
 
     painter.setOpacity(1.0);
 
-    for (auto selectedStar : m_selectedStars) {
+    for (auto selectedNode : m_selectedNodes) {
         // draw shadow of the selected node
+        const Node node = selectedNode.second;
+        const QPointF xy = QPointF(node.x() * currEdgeSize(), node.y() * currEdgeSize());
+
+        Star selectedStar(node, xy, {});
+        selectedStar.edges.reserve(node.outEdges().size());
+        for (auto const& ep : node.outEdges()) {
+            QPointF xy2 = nodePoint(ep.second.neighbour(), currEdgeSize());
+            QLineF line(xy, xy2);
+            if (!m_showNodes || line.length() - m_nodeRadius * 2. > 4.0) {
+                selectedStar.edges.push_back({ep.second, line});
+            }
+        }
+        selectedStar.edges.shrink_to_fit();
```

Comment:
it's not an efficient approach; we should not be computing things in a "draw" function
actually, we should compute the coordinates only when the object is selected (or when something changes in the view)
(note that `m_selectedStars` is not being used after your change --- why?)

Reply:
I'm still looking into this (and that's why I didn't remove ``m_selectedStars``)

While ``m_selectedNodes`` is automatically updated when moving a node, ``m_selectedStars`` is not, since it doesn't hold a pointer to the node that we're changing. That would mean that when moving a group of selected nodes, the viewport wouldn't respond to them moving. Maybe a signal could be emitted whenever there was a need to re-calculate the node coordinates which would call something like ``updateSelectedStars``

Reply:
@cardinot 
I updated it to utilize signals but I was wondering whether this is memory-safe or I need to explicitly delete the replaced star (or change its attributes in-place)